ci/authn-gcp/get_gcp_id_tokens.sh

```diff
@@ -0,0 +1,154 @@
+  fi
+  echo "-- ${COMMAND} command exists"
+}
+
+check_instance_name_arg() {
+  if [ -z "$1" ]; then
+    echo "-- GCP instance name is required. Usage: ./${PROGNAME} [GCP_INSTANCE_NAME]"
+    print_running_gcp_instances
+    error_exit "GCP instance name is required!"
+  else
+    INSTANCE_NAME=$1
+    echo "-- GCP instance name is set to: '${INSTANCE_NAME}'."
+  fi
+}
+
+#	----------------------------------------------------------------
+# Checks if the given instance name argument exist and in RUNNING
+# status, extracts the instance zone for later use;
+# exit the script with error otherwise.
+#	----------------------------------------------------------------
+ensure_instance_exists_and_running() {
+  local instance_name="$1"
+  # 'gcloud' response format: 'instance-zone;STATUS', (e.g. europe-west3-c;RUNNING).
+  local format="value[separator=';'](zone.basename(),status)"
+
+  # Get list of GCP instances in all zones filtered by exact match regex '$instance_name'.
+  # Returns 'instance-zone;STATUS'
+  # if the instance exists; empty string otherwise.
+  local instance_info=$(gcloud compute instances list --filter="name ~ ^$instance_name$" --format="$format")
+
+  if [[ "$instance_info" != "" ]]; then
+    INSTANCE_EXISTS=1
+    #	----------------------------------------------------------------
+    # The $instance_info output is in the form of 'instance-zone;STATUS',
+    # (e.g. europe-west3-c;RUNNING).
+    # Below we extract the following:
+    # - instance zone: used in the 'gcloud compute ssh' command.
+    # - instance status: used to validate that the instance is running.
+    #	----------------------------------------------------------------
+    set_instance_zone "$instance_info"
+    local instance_status=$(echo "$instance_info" | cut -f2 -d ';')
+
+    if [ "$instance_status" = "RUNNING" ]; then
+      echo "-- GCP instance '${instance_name}', zone: '$INSTANCE_ZONE', status: '$instance_status'."
+      INSTANCE_RUNNING=1
+    else
+      echo "-- GCP instance '${instance_name}', zone: '$INSTANCE_ZONE', NOT RUNNING!, status: '$instance_status'."
+      print_running_gcp_instances
+      error_exit "Instance is not running."
+    fi
+  else
+    echo "-- GCP instance '${instance_name}' NOT FOUND!"
+    print_running_gcp_instances
+    error_exit "Instance not found."
+  fi
+}
+
+set_instance_zone() {
+  local instance_info="$1"
+  INSTANCE_ZONE=$(echo "$instance_info" | cut -f1 -d ';')
+}
+
+print_running_gcp_instances() {
+  echo "-- Retrieving running instances..."
+  gcloud compute instances list --limit=50\
+    --format="table[box,title='Running Compute Instances'](name,machine_type.basename(),status)" \
+    --filter="STATUS=RUNNING"
+}
+
+get_tokens_into_files() {
+  if [ "${INSTANCE_EXISTS}" = "0" ] | [ "${INSTANCE_RUNNING}" = "0" ]; then
+    error_exit "-- Cannot run command, GCP instance '${INSTANCE_NAME}' not in a valid state!"
+  fi
+
+  get_token_into_file "full" "conjur/cucumber/host/test-app" "gcp_token_valid"
```

Comment:
consider to change the tokens files names to gce_ since we will have in the future tokens for other components

Reply:
The cli is looking for a `gcp_valid_token` when you run `./cli exec --authn-gcp`.